src/Umbraco.Cms.StaticAssets/Umbraco.Cms.StaticAssets.csproj

```diff
@@ -25,43 +17,32 @@
+  <Target Name="BuildBellePreconditions" BeforeTargets="Build">
+    <Message Text="Skip BuildBelle target because UmbracoBuild is '$(UmbracoBuild)' (this is not Visual Studio)" Importance="high" Condition="'$(UmbracoBuild)' != ''" />
+    <Message Text="Skip BuildBelle target because '$(BellePath)' already exists" Importance="high" Condition="Exists('$(BellePath)')" />
+    <Message Text="Call BuildBelle target because UmbracoBuild is empty (this is Visual Studio) and '$(BellePath)' doesn't exist" Importance="high" Condition="'$(UmbracoBuild)' == '' and !Exists('$(BellePath)')" />
+    <CallTarget Targets="BuildBelle" Condition="'$(UmbracoBuild)' == '' and !Exists('$(BellePath)')" />
   </Target>
-
-
-  <Target Name="BelleBuild">
-    <Exec WorkingDirectory="$(ProjectDir)/../Umbraco.Web.UI.Client/" Command="npm ci --no-fund --no-audit --prefer-offline" />
-    <Exec WorkingDirectory="$(ProjectDir)/../Umbraco.Web.UI.Client/" Command="npm run build:skip-tests" />
+  
+  <Target Name="BuildBelle">
+    <Exec WorkingDirectory="$(ProjectDir)..\Umbraco.Web.UI.Client\" Command="npm ci --no-fund --no-audit --prefer-offline" />
+    <Exec WorkingDirectory="$(ProjectDir)..\Umbraco.Web.UI.Client\" Command="npm run build:skip-tests" />
   </Target>
 
-  <!-- Clean Belle when cleaning and rebuilding, but only in Visual Studio -->
-  <Target Name="CleanPreconditions" AfterTargets="Clean" Condition="'$(UmbracoBuild)' == ''">
-    <Message Text="-CleanPreconditions-" Importance="high" />
-    <Message Text="Nothing to clean, as $(BellePath) does not exist." Importance="High" Condition="!Exists('$(BellePath)')" />
-    <Message Text="Not cleaning (found src/preserve.belle)." Importance="High" Condition="Exists('$(BellePath)') and Exists('$(SolutionDir)preserve.belle')" />
-    <Message Text="Remove $(BellePath)." Importance="High" Condition="Exists('$(BellePath)') and !Exists('$(SolutionDir)preserve.belle')" />
+  <Target Name="CleanBellePreconditions" AfterTargets="Clean" Condition="'$(UmbracoBuild)' == ''">
+    <Message Text="Skip CleanBelle target because '$(BellePath)' doesn't exist" Importance="high" Condition="!Exists('$(BellePath)')" />
+    <Message Text="Skip CleanBelle target because preserve.belle marker file exists" Importance="high" Condition="Exists('$(BellePath)') and Exists('$(SolutionDir)preserve.belle')" />
+    <Message Text="Call CleanBelle target because '$(BellePath)' exists and preserve.belle marker file doesn't exist" Importance="high" Condition="Exists('$(BellePath)') and !Exists('$(SolutionDir)preserve.belle')" />
+    <CallTarget Targets="CleanBelle" Condition="Exists('$(BellePath)') and !Exists('$(SolutionDir)preserve.belle')" />
+  </Target>
+  
+  <Target Name="CleanBelle">
     <ItemGroup>
       <BelleLib Include="$(BellePath)" />
     </ItemGroup>
-    <RemoveDir Directories="@(BelleLib)" Condition="Exists('$(BellePath)') and !Exists('$(SolutionDir)preserve.belle')" />
+    <RemoveDir Directories="@(BelleLib)" />
   </Target>
```

Comment:
It would make much more sense if the actual files in `Umbraco.Web.UI.Client` are moved into the `Umbraco.Cms.StaticAssets` project, so we can remove an additional project from the solution and not have this weird cross copying in place...

The NPM install can also be made optional and only executed when the `package-lock.json` file has changed (ensuring users will always have an up-to-date install) by using the following target:

```xml
<Target Name="NpmInstall" BeforeTargets="Build" Inputs="package-lock.json" Outputs="node_modules/.package-lock.json">
	<Message Text="Installing NPM packages" Importance="high" />
	<Exec Command="npm ci --no-fund --no-audit --prefer-offline" />
</Target>
```